processing/src/main/java/io/druid/query/groupby/epinephelinae/SpillingGrouper.java

```diff
@@ -99,6 +103,25 @@ public SpillingGrouper(
           limitSpec.getLimit(),
           sortHasNonGroupingFields
       );
+      // if configured buffer size is too small to support limit push down, don't apply that optmization
+      if (!limitGrouper.validateBufferCapacity(mergeBufferSize)) {
+        if (sortHasNonGroupingFields) {
+          log.info("Ignoring forceLimitPushDown, insufficient buffer capacity.");
```

Comment:
IMO, this message should be at "debug" level.

Reply:
Changed to debug